docs/guide/features.md

```diff
@@ -537,6 +537,26 @@ const modules = import.meta.glob('./dir/*.js', {
 })
 ```
 
+#### Base Path
+
+You can also use the `base` option to provide base path for the imports:
+
+```ts twoslash
+import 'vite/client'
+// ---cut---
+const moduleBase = import.meta.glob('**/*.js', {
+  base: './base',
+})
+```
+
+```ts
+// code produced by vite:
+const moduleBase = {
+  'dir/foo.js': () => import('./dir/foo.js'),
```

Comment:
You're right; it should definitely be consistent. I've made the changes.
[7f0f5e7](https://github.com/vitejs/vite/pull/18510/commits/7f0f5e754142871a9bc0431264fd04ab19cb56a1)

Reply:
I expect the key to be without `./` as `'**/*.js'` does not have `./` at the head. I think this depends on how to describe the `base` option. If we describe this as a prefix string, then I think we should remove `./` from the key. If we describe this as a base path, then I think we should add `./` to the key. A prefix string is more powerful than base path because it allows non-path prefix and maybe it works well with aliases, but the base path feels more natural with the name `base`.

Reply:
I didn't notice this PR is indirectly supporting `'**/*.js'` when `base` is set. I was thinking more like the latter, a base path to resolve the glob from and it feels easier to visualize (for me), e.g. as if you're resolving the glob in a different file path.

I think both behaviours should be equally powerful and can do the same things. But if the behaviour is more like a prefix, then perhaps the option should be called `prefix`, but then someone could ask for `suffix` too.

I also think the behaviour could be easier implemented if treated as a base path, so instead of using the file dir or root (for virtual modules), we could use whatever passed by `base` instead.

Reply:
> I think both behaviours should be equally powerful and can do the same things.

I think prefix one is a bit more powerful. For the following files,

- `foo-bar/a.js`
- `foo-baz/b.js`

if the `base` behaves as prefix, you can write `import.meta.glob('**/*.js', { base: 'foo-' })['bar/a.js']`. But if the `base` behaves as base, you cannot write that and need to write `import.meta.glob('./**/*.js', { base: './' })['foo-bar/a.js']`.

That said, I'm fine with going with base-like behavior.

Reply:
I also think we should keep base as a path and not as a prefix.

Reply:
I have modified the examples in the documentation to treat them as base instead of prefix.
[500cb14](https://github.com/vitejs/vite/pull/18510/commits/500cb145d9f3bc947387bb0c286cd0f427effc9f)